### PR TITLE
Clean up _IotMqtt_Deserialize* proof harnesses

### PR DESCRIPTION
**Pull request by feliperodri**
*Description of changes:*

- Updates the name of the proof harnesses to match the function under verification;
- Update all `cbmc-batch.yaml` files;
- Adds missing `cbmc-viewer.json` files;

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by markrtuttle** [comment]
I recommend closing this pull request.  I don't think this renaming is work the effort (and I personally prefer not having directories and files begin with an underscore).

**Comment by feliperodri** [comment]
> I recommend closing this pull request. I don't think this renaming is worth the effort to get the proofs working again (and I personally prefer not having directories and files begin with an underscore).

@markrtuttle I've updated the folder names and removed the underscores. This PR also adds missing `cbmc-viewer.json` files to the deserialized proofs as well as updates the `cbmc-batch.yaml` files, so I recommend thet we keep it.

File: cbmc/proofs/IotMqtt_DeserializeConnack/cbmc-viewer.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    "IotSemaphore_Post",
+    "IotSemaphore_TimedWait",
+    "IotSemaphore_Wait",
+    "IotTaskPool_CreateJob",
+    "IotTaskPool_GetSystemTaskPool",
+    "IotTaskPool_ScheduleDeferred",
+    "IotTaskPool_strerror",
+    "IotTaskPool_TryCancel"
+  ],
+  "proof-name": "_IotMqtt_DeserializeConnack",
```
Comment (danielsn) [inline cbmc/proofs/IotMqtt_DeserializeConnack/cbmc-viewer.json]: Why is there an _?

Comment (feliperodri) [reply cbmc/proofs/IotMqtt_DeserializeConnack/cbmc-viewer.json]: To match exactly the name of the function under verification.

**Pull request merged**